konnectivity-client/pkg/client/client.go

```diff
@@ -86,12 +86,12 @@ func CreateSingleUseGrpcTunnel(address string, opts ...grpc.DialOption) (Tunnel,
 		readTimeoutSeconds: 10,
 	}
 
-	go tunnel.serve(c)
+	go tunnel.serve(ctx, c)
 
 	return tunnel, nil
 }
 
-func (t *grpcTunnel) serve(c clientConn) {
+func (t *grpcTunnel) serve(ctx context.Context, c clientConn) {
```

Comment:
Not sure why we're bothering to add the context here if we're not planning on using it.

Reply:
if there anywhere inside this method we block that we should also short circuit on `<-ctx.Done()` (like a peer to the `case <-timer.C` case)?

Reply:
If we find one then I'm fine changing the signature but till then I don't see the need.

Reply:
I'll remove from here. I tried adding a check for <-ctx.Done() at the top of serve() for loop, but via unit test I found it unnecessary (because RecvMsg() will return an error)